### DISCUSSION TITLE
site to site vpn

### DISCUSSION THREAD

**Discussion by hwz9612** [strongswan/strongswan#2225] [Q&A]
Hi,
Now I have created a network environment as follows:
![image](https://github.com/strongswan/strongswan/assets/35334392/1337de57-c8a5-4c54-872f-a072ea52ed09)
The sun gateway is configured according to the following figure:
![image](https://github.com/strongswan/strongswan/assets/35334392/6ac36461-0d7b-4e20-9f17-d73db7ee2636)
The moon gateway is configured according to the following figure:
![image](https://github.com/strongswan/strongswan/assets/35334392/4532fa14-456c-4f3b-be33-817857a29abc)
The CA certificate of moon is generated according to the following steps:
`pki --gen --type ed25519 --outform pem > strongswanKey.pem
pki --self --ca --lifetime 3652 --in strongswanKey.pem \
           --dn "C=CH, O=strongSwan, CN=strongSwan Root CA" \
           --outform pem > strongswanCert.pem
pki --gen --type ed25519 --outform pem > moonKey.pem
pki --req --type priv --in moonKey.pem \
          --dn "C=CH, O=strongswan, CN=moon.strongswan.org" \
          --san moon.strongswan.org --outform pem > moonReq.pem
pki --issue --cacert strongswanCert.pem --cakey strongswanKey.pem \
            --type pkcs10 --in moonReq.pem --serial 01 --lifetime 1826 \
            --outform pem > moonCert.pem`
The generated CA certificate of moon is as follows:
![image](https://github.com/strongswan/strongswan/assets/35334392/e8dd6641-952e-4af7-aae9-8c71c6597871)
The CA certificate of sun is generated according to the following steps:
`pki --gen --type ed25519 --outform pem > strongswanKey.pem
pki --self --ca --lifetime 3652 --in strongswanKey.pem \
           --dn "C=CH, O=strongSwan, CN=strongSwan Root CA" \
           --outform pem > strongswanCert.pem
pki --gen --type ed25519 --outform pem > sunKey.pem
pki --req --type priv --in moonKey.pem \
          --dn "C=CH, O=strongswan, CN=sun.strongswan.org" \
          --san sun.strongswan.org --outform pem > sunnReq.pem
pki --issue --cacert strongswanCert.pem --cakey strongswanKey.pem \
            --type pkcs10 --in sunReq.pem --serial 01 --lifetime 1826 \
            --outform pem > sunCert.pem`
The generated CA certificate of sun is as follows:
![image](https://github.com/strongswan/strongswan/assets/35334392/65ecac3e-8d23-43c5-9ded-867caefb9118)
After the configuration is completed, we execute the following command on the sun gateway:
`ipsec start
swanctl --load-all`
The echo is as follows:
![image](https://github.com/strongswan/strongswan/assets/35334392/f0a0c603-ff03-41c8-9c68-a39a1fe2b854)
After the configuration is completed, we execute the following command on the moon gateway:
`ipsec start
swanctl --load-all`
The echo is as follows:
![image](https://github.com/strongswan/strongswan/assets/35334392/39ff9546-66d3-4a64-bf00-228f067ae055)
Why is this happening, and how can I solve this problem?

**Comment by tobiasbrunner** [comment]
> Why is this happening, and how can I solve this problem?

Why is what happening? What problem?

No idea if it's related, but why would you generate a CA certificate for each host? While you could do that, usually, you have a common CA that issues certificates for all involved hosts.

**Reply by hwz9612** [reply]
Hi，
According to the following figure, I can see that "unsupported Key type in '/etc/swanctl/private/moonKey.pem' ". What should I do?
![image](https://github.com/strongswan/strongswan/assets/35334392/a4632a8a-d26c-4daf-82ab-3a83c2e11f1d)

**Reply by tobiasbrunner** [reply]
But you also see that the key is loaded successfully right afterwards. Maybe you configured something incorrectly about the key in swanctl.conf.
